docs/proposals/dbaas/01-architecture/proposal.md

```diff
@@ -0,0 +1,465 @@
+
+```mermaid
+sequenceDiagram
+    autonumber
+    PMM UI->>ManageD: Add kubernetes cluster
+    ManageD->>K8S: Install OLM
+    break when olm is installed 
+    	K8S->>OLM: provisions an OLM
+    end
+    Note right of K8S: OLM is provisioned!
+    ManageD->>OLM: Install required operators and create a manual subscription 
+    OLM->>Catalog: Gets the latest version of the operator 
+    loop when subscription is created 
+        OLM->>Catalog: manages subscription and checks for a new version
+    end
+    Note right of OLM: A new version of operator is available
+    OLM->>Catalog: Downloads and installs a new version of operator and waiting for a user approval to switch to a new version.
+    PMM UI->>ManageD: Check for operator update
+    ManageD->>OLM: Is there an update available?
+    OLM->>ManageD: Sure. I downloaded it for you and waiting for your approval to switch to a new version
+    ManageD->>PMM UI: Yeah, there's a new version of operator available
+    Note left of PMM UI: A user approved the upgrade request
+    PMM UI->> ManageD: approve a new version of operator
+    ManageD->>OLM: approve an subscription upgrade request
+    Note left of OLM: A new version of operator is available 
+```    
+This process makes version service usage for supported versions of operators obsolete because it gets the latest version of available operator from the catalog. However, the catalog can be managed by our team and we'll push only supported versions of operators to the catalog. This also will improve integration testing for unreleased versions of operators.
+
+### Working with the databases
+
+The second part of the proposal is implementing the generic and simplified API to create database clusters via k8s operator. The proposal discusses the drawbacks of the current implementation in the Motivation section and moving to the dbaas-operator will have the following benefits 
+
+1. Kubernetes native way to work with database clusters via unified and general specifications.
+2. PMM/DBaaS does need to care about the state of a managed database.
+3. PMM/DBaaS should manage clusters that were created via PMM/DBaaS.Yet,  it manages clusters that were created via Percona operators now. Dbaas-operator can use its own kubernetes annotations to specify database clusters managed by PMM/DBaaS. Also, it opens room for additional features such as CR templates to create a database cluster and update a version of a template using the same annotations spec.
+4. Installation and upgrading of dbaas-operator will be managed by OLM as proposed above.
+5. Dbaas-operator opens a possibility to create an easy to use integration testing framework using codecept.js/playwright or even simple bash scripts. Integration testing can be covered by development team. It can use Github actions as a pipeline to run additional tests to check create/load/run queries agains exposed/non-exposed database clusters. 
```

Comment:
```suggestion
5. Dbaas-operator opens a possibility to create an easy-to-use integration testing framework using codecept.js/playwright or even a simple bash scripts. Integration testing can be covered by the development team. It can use Github actions as a pipeline to run additional tests to check create/load/run queries agains exposed/non-exposed database clusters. 
```